### PR TITLE
Add user normalization dictionary

### LINKED ISSUE #115: Add user synonym entries dictionary
We got an idea from one of our users that uses word synonyms in the Ivona app to improve the prununciation for utterances the app doesn't yet know or where the user wants to deviate from what the app decides.

- [ ] Add prununciation dictionary with alternative pronunciation

### PR DESCRIPTION
**Pull request by lumpidu**
Add user normalization Dictionary.

The user can add normalization entries for an alternative pronunciation of words/tokens. These alternative pronunciations will have precedence over the built-in normalization rules and will apply the given replacements of one or multiple words for any such found terms in the read text.

To make usage easier, we simply make replacements possible on a grapheme level without exposing regular expression syntax. The user can immediately listen to the entered term and replacement text with the current voice via play buttons.

By default, the user normalization dictionary is empty. Currently, it's not possible to import or export the dictionary.

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -86,4 +90,10 @@
     <string name="voice_deletion_q">Really delete voice?</string>
     <string name="voice_deletion_not_possible">You cannot delete the voice you are currently using, please select a different voice and try again</string>
     <string name="no_email_client">There is no email client installed</string>
+    <string name="term_title">Term</string>
```
Comment (bnika) [inline app/src/main/res/values/strings.xml:93]: orð

File: app/src/main/res/values-is-rIS/strings.xml [hunk 2]
```diff
@@ -21,6 +22,8 @@
     <string name="voice_type">Tegund</string>
     <string name="speak_voice_title">Spila</string>
     <string name="play_voice_title_description">Lesa sýndan texta</string>
+    <string name="play_term_title_description">Lesa sýndan texta</string>
+    <string name="play_replacement_title_description">Lesa sýndan texta</string>
```
Comment (bnika) [inline app/src/main/res/values-is-rIS/strings.xml:26]: only: Lesa texta

File: app/src/main/res/values-is-rIS/strings.xml [hunk 3]
```diff
@@ -31,6 +34,7 @@
     <string name="en_US">Enska</string>
     <string name="ok">Í lagi</string>
     <string name="cancel">Hætta</string>
+    <string name="delete_norm_entry">Delete</string>
```
Comment (bnika) [inline app/src/main/res/values-is-rIS/strings.xml]: Eyða

File: app/src/main/res/values-is-rIS/strings.xml [hunk 4]
```diff
@@ -86,4 +90,10 @@
     <string name="voice_deletion_q">Virkilega eyða rödd?</string>
     <string name="voice_deletion_not_possible">Þú getur ekki eytt röddinni sem er valin, vinsamlegast veldu aðra rödd og reyndu aftur</string>
     <string name="no_email_client">Það er ekkert tölvupóstforrit uppsett</string>
+    <string name="term_title">Term</string>
+    <string name="replacement_title">Replacement</string>
+    <string name="save_entry">Save</string>
+    <string name="add_dict_entry">Add entry</string>
+    <string name="dict_entry_deletion_title">Delete entry</string>
+    <string name="dict_entry_deletion_q">Really delete entry?</string>
```
Comment (bnika) [inline app/src/main/res/values-is-rIS/strings.xml]: Term: Orð
Replacement: Lesa sem
Save: Vista
Add entry: Bæta við færslu
Delete entry: Eyða færslu
Really delete entry?: Ertu viss um að þú viljir eyða færslu?

Comment (lumpidu) [reply app/src/main/res/values-is-rIS/strings.xml]: done

File: app/src/main/res/values-is-rIS/strings.xml (hunk 3 above)
Comment (lumpidu) [reply app/src/main/res/values-is-rIS/strings.xml]: done

File: app/src/main/res/values-is-rIS/strings.xml (hunk 2 above)
Comment (lumpidu) [reply app/src/main/res/values-is-rIS/strings.xml:26]: done

**Review by bnika — APPROVED** [review]

**Pull request merged**